components/collector/tests/source_collectors/dependency_track/test_dependencies.py

```diff
@@ -50,26 +50,51 @@ async def test_no_projects(self):
-        self.assert_measurement(response, value="1", entities=self.create_entities("1.1", "update possible"))
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.dependencies("1.1")])
+        self.assert_measurement(response, value="1", entities=self.entities("1.1", "update possible"))
 
     async def test_up_to_date_dependencies(self):
         """Test one project with a component that is up to date."""
-        response = await self.collect(get_request_json_side_effect=[self.projects, self.create_dependencies("1.0")])
-        self.assert_measurement(response, value="1", entities=self.create_entities("1.0", "up-to-date"))
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.dependencies("1.0")])
+        self.assert_measurement(response, value="1", entities=self.entities("1.0", "up-to-date"))
 
     async def test_unknown_latest_version(self):
         """Test one project with a component whose latest version is unknown."""
-        response = await self.collect(get_request_json_side_effect=[self.projects, self.create_dependencies("")])
-        self.assert_measurement(response, value="1", entities=self.create_entities("unknown", "unknown"))
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.dependencies("")])
+        self.assert_measurement(response, value="1", entities=self.entities("unknown", "unknown"))
 
     async def test_filter_by_latest_version_status(self):
         """Test that components can be filtered by latest version status."""
         self.set_source_parameter("latest_version_status", ["update possible"])
-        response = await self.collect(get_request_json_side_effect=[self.projects, self.create_dependencies("1.0")])
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.dependencies("1.0")])
         self.assert_measurement(response, value="0", entities=[])
+
+    async def test_filter_by_project_name(self):
+        """Test filtering projects by name."""
+        self.set_source_parameter("project_names", ["other project"])
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.dependencies("1.0")])
+        self.assert_measurement(response, value="0", entities=[])
+
+    async def test_filter_by_project_regular_expression(self):
+        """Test filtering projects by regular expression."""
+        self.set_source_parameter("project_names", ["project .*"])
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.dependencies("1.0")])
+        self.assert_measurement(response, value="1", entities=self.entities("1.0", "up-to-date"))
+
+    async def test_filter_by_project_version(self):
+        """Test filtering projects by version."""
+        self.set_source_parameter("project_versions", ["1.2", "1.3"])
```

Comment:
Sneaky version numbers containing dots are also regex

Reply:
You're right. Hmm. That means "1.2.3" would match e.g. "102.3" and "1.203". I guess it's unlikely in practice, but we could change `is_regexp` to prevent interpreting `\d+\.\d+(\.\d)*` as regular expression? What do you think?

Reply:
I think we should keep it as-is, but perhaps a story on the backlog to think about special chars such as dot in regex. There are other solutions such as a tooltip, to inform the user that their input might lead to unexpected results